Lib/test/test_operator.py

```diff
@@ -474,6 +474,30 @@ def __getitem__(self, other): return 5  # so that C is a sequence
         self.assertEqual(operator.ixor     (c, 5), "ixor")
         self.assertEqual(operator.iconcat  (c, c), "iadd")
 
+    def test_iconcat_without_getitem(self):
+        operator = self.module
+        class X: pass
+
+        msg = "'X' object can't be concatenated"
+        with self.assertRaisesRegex(TypeError, msg):
+            operator.iconcat(X(), X())
+
+    def test_index(self):
```

Comment:
`operator.index` is already tested in `Lib/test/test_index.py`.

Reply:
I guess test_index only checks the C version of the module. This test covers the Python version with `self.module` used for C and Python version of the module.

Reply:
Agree. Although it would be nice to extend this test. At least test it with int, float, Fraction, Decimal, None.